src/Sainsburys/HttpService/Components/Middlewares/MiddlewareLibrary/BeforeMiddleware/UnpackJsonParametersToArray.php

```diff
@@ -0,0 +1,59 @@
+<?php
+namespace Sainsburys\HttpService\Components\Middlewares\MiddlewareLibrary\BeforeMiddleware;
+
+use Sainsburys\HttpService\Components\Middlewares\Http\RequestAndResponse;
+use Sainsburys\HttpService\Components\Middlewares\MiddlewareInterfaces\BeforeMiddleware;
+use Sainsburys\HttpService\Components\ErrorHandling\Exceptions\WithStatusCode\MalformedJsonRequest;
+
+class UnpackJsonParametersToArray implements BeforeMiddleware
+{
+
+    private $originalParams;
+
+    private $proxyInjectedParams;
+
+    public function __construct(string $originalParams, string $proxyInjectedParams)
```

Comment:
What I meant was that this should be an array of keys, so this class has no knowledge of HARA